common/buildcraft/core/recipes/FlexibleRecipe.java

```diff
@@ -145,7 +152,19 @@ public void setContents(String iid, Object iioutput, int iEnergyCost, long iCraf
 			} else if (i instanceof List) {
 				inputItemsWithAlternatives.add((List) i);
 			} else if (i instanceof String) {
-				inputItemsWithAlternatives.add(OreDictionary.getOres((String) i));
+				if (index + 1 >= input.length) {
+					inputItemsWithAlternatives.add(OreDictionary.getOres((String) i));
+				} else if (input[index + 1] instanceof Integer) {
+					skip = true;
```

Comment:
Could you please simply use `index++;` here instead of `skip = true;`? That's much less hacky, and you can do this with a for loop.

Reply:
I did, but the style checker complained aborted the build, I can see if there is an annotation for it to ignore that method though.

Reply:
It's 1.7.10, nobody supports it anyway, I wouldn't normally advise this but just throw out the style checker and Javadoc if it's causing problems... unless @AlexIIL wants to backport some improvements from the new branches.